### ISSUE TITLE
Fails on some binary files

### ISSUE THREAD

**Issue by skliper** [OpenC3/openc3-cosmos-cfdp#4]
Receive failure when receiving (UNACK) binary files:
```
UndefinedConversionError : "\xC0" from ASCII-8BIT to UTF-8
/usr/lib/ruby/3.2.0/delegate.rb:349:in `write'
/usr/lib/ruby/3.2.0/delegate.rb:349:in `block in delegating_block'
/tmp/d20230914-15-52ebm9/app/models/cfdp_receive_transaction.rb:580:in `handle_pdu'
/tmp/d20230914-15-52ebm9/app/models/cfdp_user.rb:74:in `block (2 levels) in start'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/utilities/store_autoload.rb:153:in `block (3 levels) in read_topics'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/utilities/store_autoload.rb:151:in `each'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/utilities/store_autoload.rb:151:in `block (2 levels) in read_topics'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/utilities/store_autoload.rb:150:in `each'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/utilities/store_autoload.rb:150:in `block in read_topics'
/usr/lib/ruby/gems/3.2.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:110:in `block (2 levels) in with'
/usr/lib/ruby/gems/3.2.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:109:in `handle_interrupt'
/usr/lib/ruby/gems/3.2.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:109:in `block in with'
/usr/lib/ruby/gems/3.2.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:106:in `handle_interrupt'
/usr/lib/ruby/gems/3.2.0/gems/connection_pool-2.4.1/lib/connection_pool.rb:106:in `with'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/utilities/store_autoload.rb:146:in `read_topics'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/utilities/store_autoload.rb:58:in `public_send'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/utilities/store_autoload.rb:58:in `method_missing'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/topics/topic.rb:29:in `public_send'
/usr/lib/ruby/gems/3.2.0/gems/openc3-5.10.1/lib/openc3/topics/topic.rb:29:in `method_missing'
/tmp/d20230914-15-52ebm9/app/models/cfdp_user.rb:54:in `block in start'
```

This case traces to using the default `w+` here:
https://github.com/OpenC3/openc3-cosmos-cfdp/blob/fa4b65f1928dd54ceeeb074f021b97f2711757b0/microservices/CFDP/app/models/cfdp_receive_transaction.rb#L557

Looks like there's a few other uses that default to `w+` but I haven't dug into which ones matter.

**Comment by skliper** [comment]
Fixed w/ `@tmp_file ||= Tempfile.new('cfdp', :encoding => 'binary')`, at least for our use case.

**Comment by ryanmelt** [comment]
closed by 0994d0e
